### PR TITLE
Improve doc strings

### LINKED ISSUE #84: Improve docstrings
Docstrings could generally be improved, via:

* Description of return values
* Examples

### PR DESCRIPTION
**Pull request by KellyLBennett**
Need more information on locate_region, open functions that return zarr, site_annotations.
aa_allele_frequencies not yet supported in datalab.

Resolves #84.

### CODE REVIEW

**Comment by KellyLBennett** [comment]
@alimanfoo What does the site annotations function do please? This seems to return integers of -1, 0 and 1 but I am not sure how this relates to "codon position", "codon degeneracy" etc.

File: malariagen_data/ag3.py [hunk 1]
```diff
@@ -691,7 +692,10 @@ def snp_sites(
 
         Returns
         -------
-        d : dask.array.Array or tuple of dask.array.Array
+        pos, ref, alt : dask.array.Array
```
Comment (alimanfoo) [inline malariagen_data/ag3.py:695]:
```suggestion
        d : dask.array.Array
```

File: malariagen_data/ag3.py [hunk 2]
```diff
@@ -860,7 +865,7 @@ def open_genome(self):
 
         Returns
         -------
-        root : zarr.hierarchy.Group
+        root : zarr.hierarchy.Group of zarr.core.Array for the available chromosome regions.
```
Comment (alimanfoo) [inline malariagen_data/ag3.py:868]:
```suggestion
        root : zarr.hierarchy.Group
            Zarr hierarchy containing the reference genome sequence.
```

File: malariagen_data/ag3.py [hunk 3]
```diff
@@ -887,6 +892,7 @@ def genome_sequence(self, region, inline_array=True, chunks="native"):
         Returns
         -------
         d : dask.array.Array
+            An array of a specified region of the reference genome sequence.
```
Comment (alimanfoo) [inline malariagen_data/ag3.py:895]:
```suggestion
            An array of nucleotides giving the reference genome sequence for the given contig.
```

File: malariagen_data/ag3.py [hunk 4]
```diff
@@ -1054,6 +1062,7 @@ def snp_effects(
         Returns
         -------
         df : pandas.DataFrame
+            A dataframe of all possible variants and their effects, one row per variant.
```
Comment (alimanfoo) [inline malariagen_data/ag3.py:1065]:
```suggestion
            A dataframe of all possible SNP variants and their effects, one row per variant.
```

File: malariagen_data/ag3.py [hunk 5]
```diff
@@ -1475,7 +1487,8 @@ def snp_calls(
 
         Returns
         -------
-        ds : xarray.Dataset
+        ds : xarray.Dataset of xarray.DataArray
+            An array dataset of SNP sites, site filters and genotype calls.
```
Comment (alimanfoo) [inline malariagen_data/ag3.py:1491]:
```suggestion
        ds : xarray.Dataset
            An dataset containing SNP sites, site filters and genotype calls.
```

File: malariagen_data/ag3.py [hunk 6]
```diff
@@ -1646,12 +1659,11 @@ def cnv_hmm(
 
         Returns
         -------
-        ds : xarray.Dataset
+        ds : xarray.Dataset of xarray.DataArray
+            An array dataset of CNV HMM calls and associated data.
```
Comment (alimanfoo) [inline malariagen_data/ag3.py:1663]:
```suggestion
        ds : xarray.Dataset
            A dataset of CNV HMM calls and associated data.
```

File: malariagen_data/ag3.py [hunk 7]
```diff
@@ -1732,7 +1744,8 @@ def cnv_coverage_calls(
 
         Returns
         -------
-        ds : xarray.Dataset
+        ds : xarray.Dataset of xarray.DataArray
+            An array dataset of CNV HMM breakpoints and their associated data.
```
Comment (alimanfoo) [inline malariagen_data/ag3.py:1748]:
```suggestion
        ds : xarray.Dataset
            A dataset of CNV alleles and genotypes.
```

File: malariagen_data/ag3.py [hunk 8]
```diff
@@ -1936,12 +1949,11 @@ def cnv_discordant_read_calls(
 
         Returns
         -------
-        ds : xarray.Dataset
+        ds : xarray.Dataset of xarray.DataArray
+            An array dataset of discordant read calls and associated data.
```
Comment (alimanfoo) [inline malariagen_data/ag3.py:1953]:
```suggestion
        ds : xarray.Dataset
            A dataset of CNV alleles and genotypes.
```

File: malariagen_data/ag3.py [hunk 9]
```diff
@@ -1979,7 +1991,8 @@ def gene_cnv(self, contig, sample_sets=None):
 
         Returns
         -------
-        ds : xarray.Dataset
+        ds : xarray.Dataset of xarray.DataArray
+            An array dataset of modal copy number per gene and associated data.
```
Comment (alimanfoo) [inline malariagen_data/ag3.py:1995]:
```suggestion
        ds : xarray.Dataset
            A dataset of modal copy number per gene and associated data.
```

File: malariagen_data/ag3.py [hunk 10]
```diff
@@ -2076,6 +2089,7 @@ def gene_cnv_frequencies(
         Returns
         -------
         df : pandas.DataFrame
+            A dataframe of CNV amplification and deletion frequencies in the specified cohorts, one row per gene.
```
Comment (alimanfoo) [inline malariagen_data/ag3.py:2092]:
```suggestion
            A dataframe of CNV amplification (amp) and deletion (del) frequencies in the specified cohorts, 
            one row per gene and CNV type (amp/del).
```

File: malariagen_data/ag3.py [hunk 11]
```diff
@@ -2300,7 +2314,8 @@ def haplotypes(
 
         Returns
         -------
-        ds : xarray.Dataset
+        ds : xarray.Dataset of xarray.DataArray
+            An array dataset of haplotypes and associated data.
```
Comment (alimanfoo) [inline malariagen_data/ag3.py:2318]:
```suggestion
        ds : xarray.Dataset
            A dataset of haplotypes and associated data.
```

File: malariagen_data/ag3.py [hunk 12]
```diff
@@ -1476,6 +1489,7 @@ def snp_calls(
         Returns
         -------
         ds : xarray.Dataset
+            An dataset containing SNP sites, site filters and genotype calls.
```
Comment (alimanfoo) [inline malariagen_data/ag3.py:1492]:
```suggestion
            A dataset containing SNP sites, site filters and genotype calls.
```

**Review by alimanfoo — APPROVED** [review]

**Pull request merged**